### ISSUE TITLE
False positive in PSR-7

### ISSUE THREAD

**Issue by danielspk** [kalessil/phpinspectionsea#1618]
The following code returns a false positive:

```php
use Psr\Http\Message\ResponseInterface;
use Psr\Http\Message\ServerRequestInterface;
use Psr\Http\Server\RequestHandlerInterface;

class Demo
{
    public function process(ServerRequestInterface $request, RequestHandlerInterface $handler): ResponseInterface
    {
        $request = $request->withUri($uri->withPath('/'));

        return $handler->handle($request);
    }
}
```

The `ServerRequestInterface` class extends from` RequestInterface`

When invoking the `withUri` method from a` ServerRequestInterface` type, shouldn't the `static` return be interpreted as a` ServerRequestInterface`?
